pennylane/math/quantum.py

```diff
@@ -1125,3 +1125,89 @@ def _compute_max_entropy(density_matrix, base):
     maximum_entropy = qml.math.log(rank) / div_base
 
     return maximum_entropy
+
+
+def min_entropy(state, indices, base=None, check_state=False, c_dtype="complex128"):
+    r"""Compute the minimum entropy from a state vector or density matrix on a given subsystem. It supports all
+    interfaces (Numpy, Autograd, Torch, Tensorflow and Jax).
+
+    .. math::
+        S_{\text{min}}( \rho ) = -\log( \max_{i} ( p_{i} ))
+
+    Args:
+        state (tensor_like): ``(2**N)`` state vector or ``(2**N, 2**N)`` density matrix.

```

Comment:
Can we make this accept just density matrices? See #4322.

Reply:
The code itself might not have to change much, but it would be good to add a test for batching.

Reply:
Hi @trbromley and thank you for the review! Do you have any suggestion about how to use batching with density matrices? Tests for similar functions in `quantum.py` don't seem to use it; not even `test_density_matrices_max_entropy` (A similar test for the min-entropy is already present).

Reply:
added some details in my latest comment below.